MantidPlot/src/Mantid/InstrumentWidget/InstrumentWindow.cpp

```diff
@@ -839,8 +60,8 @@ void InstrumentWindow::preDeleteHandle(
   Mantid::API::IPeaksWorkspace_sptr pws =
       boost::dynamic_pointer_cast<Mantid::API::IPeaksWorkspace>(workspace_ptr);
   if (pws) {
-    getSurface()->deletePeaksWorkspace(pws);
-    updateInstrumentView();
+    m_instrumentWidget->getSurface()->deletePeaksWorkspace(pws);
```

Comment:
The only reason you run getSurface is to delete the PeaksWorkspace. Instead of `m_instrumentWidget->getSurface()->deletePeaksWorkspace(pws)` how about the client being able to write `m_instrumentWidget->deletePeaksWorkspace(pws)` so that the surface stuff is handled internally.